File: LeetCode/C#/804. Unique Morse Code Words.cs
public class Solution {
    public int UniqueMorseRepresentations(string[] words) {
        var morseCode = new string[]{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        
        var transformationList = new List<string>();
        
        for(var i = 0; i < words.Length; i++)
        {
            var transformationString = "";
            
            foreach(var letter in words[i])
            {
                transformationString += morseCode[(int)letter - 97];
            }
            
            if(!transformationList.Contains(transformationString))
            {
                transformationList.Add(transformationString);
            }
        }
        
        return transformationList.Count;
    }
}

File: LeetCode/C++/804. Unique Morse Code Words.cpp
class Solution
{
public:
    int uniqueMorseRepresentations(vector<string> &words)
    {
        set<string> wordSet;

        for (string word : words)
        {
            string morseCode = convertToMorse(word);

            if (wordSet.find(morseCode) == wordSet.end())
            {
                wordSet.insert(morseCode);
            }
        }

        return wordSet.size();
    }

    string convertToMorse(string word)
    {
        string morseCode;
        vector<string> morseCodes = {".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."};

        for (char c : word)
        {
            int idx = int(c) - 97;
            morseCode += morseCodes[idx];
        }

        return morseCode;
    }
};